scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -199,6 +199,24 @@ sealed trait Execution[+T] extends java.io.Serializable { self: Product =>
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
 
+  /**
+   * Add an extra config for current execution.
+   * This changes of config are isolated and affect only current execution.
+   * All changes will disappear for any next operation on execution.
+   */
+  def addConfig(extraConfig: Config): Execution[T] =
+    Execution.withConfig(this) { config =>
+      config ++ extraConfig
+    }
+
+  /**
+   * Replace current the config to a new one.
+   * This changes of config are isolated and affect only current execution.
+   * All changes will disappear for any next operation on execution.
+   */
+  def withConfig(config: Config): Execution[T] =
```

Comment:
This one I really don't think we need. It has the same name as `Execution.withConfig` but it's very different. I don't think we ever really want to ignore the old config. Can you think of a use case where this is necessary?